js_modules/dagster-ui/packages/ui-core/src/assets/AssetNodeOverview.tsx

```diff
@@ -81,34 +84,50 @@ import {buildRepoAddress} from '../workspace/buildRepoAddress';
 import {workspacePathFromAddress} from '../workspace/workspacePath';
 
 export const AssetNodeOverview = ({
+  assetKey,
   assetNode,
   upstream,
   downstream,
   liveData,
   dependsOnSelf,
 }: {
-  assetNode: AssetNodeDefinitionFragment;
+  assetKey: AssetKey;
+  assetNode: AssetNodeDefinitionFragment | undefined | null;
   upstream: AssetNodeForGraphQueryFragment[] | null;
   downstream: AssetNodeForGraphQueryFragment[] | null;
   liveData: LiveDataForNode | undefined;
   dependsOnSelf: boolean;
 }) => {
-  const repoAddress = buildRepoAddress(
-    assetNode.repository.name,
-    assetNode.repository.location.name,
-  );
+  const repoAddress = assetNode
+    ? buildRepoAddress(assetNode.repository.name, assetNode.repository.location.name)
+    : buildRepoAddress('', '');
   const location = useRepositoryLocationForAddress(repoAddress);
 
   const {assetType, assetMetadata} = metadataForAssetNode(assetNode);
   const {schedules, sensors} = useMemo(() => insitigatorsByType(assetNode), [assetNode]);
-  const configType = assetNode.configField?.configType;
+  const configType = assetNode?.configField?.configType;
   const assetConfigSchema = configType && configType.key !== 'Any' ? configType : null;
-  const visibleJobNames = assetNode.jobNames.filter((jobName) => !isHiddenAssetGroupJob(jobName));
+  const visibleJobNames =
+    assetNode?.jobNames.filter((jobName) => !isHiddenAssetGroupJob(jobName)) || [];
 
   const assetNodeLoadTimestamp = location ? location.updatedTimestamp * 1000 : undefined;
 
+  const [cachedAsset, setCachedAsset] = useState<AssetTableFragment | undefined>();
+  useCachedAssets({
+    onAssets: useCallback(
+      (data) => {
+        for (const asset of data) {
+          if (JSON.stringify(asset.key.path) === JSON.stringify(assetKey.path)) {
+            setCachedAsset(asset);
```

Comment:
nit: this should only hit once right? return after?

Reply:
ah yeah

Reply:
ah yeah, that was the reason I made it a for loop instead of using `forEach` but then I forgot to actually do the return 😅